src/compute-client/src/controller.rs

```diff
@@ -216,6 +216,17 @@ impl<T> ComputeController<T> {
         })
     }
 
+    /// Return a reference-less snapshot to the indicated compute instance.
+    pub fn instance_snapshot(
+        &self,
+        id: ComputeInstanceId,
+    ) -> Result<ComputeInstanceSnapshot, InstanceMissing> {
+        self.instance(id).map(|instance| ComputeInstanceSnapshot {
+            instance_id: id,
+            collections: BTreeSet::from_iter(instance.collections_iter().map(|(id, _state)| *id)),
+        })
+    }
```

Comment:
Is there a reason why this has to live in the compute controller? I'd like to keep the compute controller API as small as possible. In this case it seems like client code already has access to all the available information, since we just collect the IDs of existing collections. So this method and the `ComputeInstanceSnapshot` type could live in adapter, no?

Reply:
Moved as much as I could into adapter. Had to add a method on instance ref.